### ISSUE TITLE
Add Performance and Resource usage KPI checks in the CI/CD pipeline

### ISSUE THREAD

**Issue by krucod3** [eclipse-ankaios/ankaios#128]
## Description

Currently we do not monitor if the resource usage or the performance of Ankaios was degraded by a commit or a PR. With time we could get into the situation where Ankaios is not lean and low-profile enough for the target use-case.

To prevent this, we can add automated KPI checks for the performance and the resource usage that are executed as part of the CI pipeline.

## Goals

Monitor the performance and resource usage of Ankaios to prevent gradual degradation.

### Summary

**To be filled when the final solution is sketched.**

### Tasks

- [ ] Task 1
- [ ] Task 2
- [ ] ...

**Comment by krucod3** [comment]
We could connect real HW as an external runner and check in a nightly build if the performance (startup time) and the resource usage (RAM, CPU) is getting worst.

**Comment by krucod3** [comment]
We should also do a manual measurement as a start and see how much CPU and memory is used by Ankaios when handling 100 container.
As a second step we should measure the time it takes to start the 100 containers and automate these steps.

Let's proceed as follows:
1. first make a manual measurement
1. make a concept how the automation can be done (maybe we need code instrumentation here too)
1. implement the checks (maybe in a separate issue)

**Comment by krucod3** [comment]
I tested triggering the start of 100 containers, here are results:

| tool | try one | try two | try three | average time |
| --- | --- | --- | --- | --- |
| Ankaios with one agent on the same node | 5,25578282 | 7,852428348 | 7,175706493 | 6,761305887 |
| bash script starting detached containers in parallel | 9,635978355 | 4,563347779 | 8,325607941 | 7,5083113583 |

As it can be observed, the results are fluctuating highly. I would argue that Podman itself needs some time to trigger the creation and gets some locks somewhere which cloud cause some delays. 

What we definitely see is that (considering statistical fluctuations) Ankaios does not add a significant overhead when starting the containers although it is keeping internal data about them and is monitoring them.

Note:
This was just a basic manual test done on my dev machine (equipped with an i7-11850H).
More sophisticated test must be done to properly measure, but at least the test gives an initial impression about the added overhead, or more like the absence of an observable overhead added by Ankaios.
